src/ngx_postgres_rewrite.c

```diff
@@ -187,7 +187,7 @@
         url_variable.len = 0;
         //fprintf(stdout, "something here %s\n", p);
 
-        while(url_variable.len < (redirect + size) - (p + 1)) {
+        while((int)url_variable.len < (redirect + size) - (p + 1)) {
```

Comment:
Well the other side of the <, (redirect + size) - (p + 1), is an int expression (not unsigned int). Therefore compiler warns that you are comparing signed and unsigned values.